src/components/NavBar/NavBar.tsx

```diff
@@ -70,12 +75,20 @@ const NavBar = () => {
             ))}
           </List>
           {/*/ Update the button to post action*/}
-          <Button sx={styles.postButton} variant="contained" type="submit">
+          <Button
+            sx={styles.postButton}
+            variant="contained"
+            type="button"
+            onClick={() => setOpenModal(!openModal)}
+          >
             Post
           </Button>
           <UserAvatar />
         </Stack>
       </Toolbar>
+      <PostButtonModal openModal={openModal} setOpenModal={setOpenModal}>
+        <ComposePost />
```

Comment:
I like that you're trying to reuse our components, but dialogs have a `DialogActions` component that we can put our post button, emoji upload button, etc inside. I think the styling might look better if we try to use the component that they have

Reply:
It seems redundant to be remaking `ComposePost`, along with the post request and whatnot

Reply:
On second thought, I agree. I think the padding still looks kind of bad though. Can we pair program and go through the styling together?